logstash-scheduler/src/main/java/org/apache/mesos/logstash/scheduler/OfferStrategy.java

```diff
@@ -62,76 +64,71 @@ public OfferResult evaluate(ClusterState clusterState, Protos.Offer offer) {
      * Offer result
      */
     public static class OfferResult {
-        final boolean acceptable;
-        final Optional<String> reason;
+        final List<String> complaints;
 
-        private OfferResult(boolean acceptable, Optional<String> reason) {
-            this.acceptable = acceptable;
-            this.reason = reason;
+        private OfferResult(List<String> complaints) {
+            this.complaints = complaints;
+        }
+
+        public boolean acceptable() {
+            return complaints.isEmpty();
```

Comment:
You could model this with `Collections.singletonList("unknown reason")` if absolutely necessary, but I find it hard to imagine a situation which would call for it